### DISCUSSION TITLE
0525 피어세션일지

### DISCUSSION THREAD

**Discussion by gvsteve24** [bcaitech1/p4-dkt-freshtomato#12] [General]
# 오늘의 TMI + EMOJI

* (민용) :cut_of_meat:
* (진현) :billed_cap:
* (아라) :sushi:
* (유라) :metamong:
* (재훈) :dizzy_face:
* (정현) :sob:

# 회의/이슈/토론

* ([이슈](https://www.daleseo.com/python-black/)) 코드 Formatter 블랙!
* ([이슈](https://github.com/bcaitech1/p4-dkt-freshtomato/issues/6)) Data Split 할때 어떻게 하는 것이 합리적인가?
  * Train, Test를 나누는데 분포기준을 잡는것
    * userId 기준으로 나누긴 하지만 분포 특성을 이야기하는 것!
  * (민용) 이야기 나온 것들의 EDA를 하면 될거같다. 시험 자체의 난이도도 있으면 괜찮을 것 같다.
    * (아라) [토론글](http://boostcamp.stages.ai/competitions/42/discussion/post/346)에 보니 난이도에 관련된 이야기도 있었음 - **testId의 앞자리 3개**
  * (아라) 각자 하나씩 맡아서 해보자! **모델은 LSTM(baseline)으로 돌리고 시드는 406으로**
    * (1번)사람으로 일단 나눈 뒤에 여기서 맨 마지막 문제를 풀었는지에 따라 다시 비율을 일정하게 train과 valid로 나눔
    * (2번)개인별로 정답률을 구한 뒤에 threshold를 이용해서 label을 다시 정의한 뒤 일정하게 나눔
      EDA에서 문제를 얼마나 많이 풀어봤냐에 따라 정답률이 증가하는 것처럼 보였음
    * (3번)개인별로 푼 시험의 개수에 따라 균일하게 나눠보면 어떨지
    * (4번)개인별로 푼 시험의 개수처럼 유사하게 문제의 개수로 나눠보면 어떨지
      -연속형변수를 어떻게 나누면 좋을지
    * (5번)knowledgetag를 얼마나 다양하게 풀었는지, 얼마나 하나에 대해 많이 풀었는지, 종류 등에 대해 다양하게 시도해볼 수 있을 것 같음
  * (아라) baseline 모델에서 split에 사용된 ratio를 9:1로 바꾸니까 좋아졌다.
  * 현재 돌린 결과를 살리기 위해서는 valid set을 나누는 seed는 변경하지 않고(default: 0), 기준만 바꿔서 수행 
    * 모델 seed는 406으로 쓰고 각자 실험 돌리기
    * 가장 좋은 기준을 찾는다면 valid set을 나누는 seed도 이때 다시 실험
    
* (컴피티션) 제출 10회중 각자 1번은 자유롭게 ! 나머지 4번은 계획짠걸로 제출하자.
* (멘토링) 금요일 3시<s>로 말씀 드릴 예정</s>
* (코드) Json으로 설정한 arguments들 동작하는 것 내일 피어세션 공유예정 
* (데이터) 테스트 데이터 셋을 학습에 이용해도 괜찮은건가?
  * Pseudolabeling 포함 데이터를 학습에 포함시켜도 된다.
  * 매뉴얼하게 입력하는 것만 아니면 괜찮다.
* (유라) 테스트 데이터셋에서 answercode -1 제외하고 학습을 돌려봄. LSTM에서 결과가 더 좋게 나왔다.
* (아라) LGBM, LSTM+attention, BERT등을 돌려봤는데 결과.
  * lgbm > LSTM > LSTMATTN >>> Bert 순
  * (민용)LGBM이 피처엔지니어링이 조금 들어가서 괜찮은 결과같다.
  * 따라서 현재 진행할 실험은 Baseline으로 돌려봐도 괜찮을 거 같다.  
* (민용) dev 브랜치에 올리는 거좀 보여줄 수 있는지
  * (아라) train.py에서  `args.model_dir`를  '{args.model_dir}/{args.wandb_run_name}' 으로 사용
  * 하고 시연함! 깃헙 으로 보여줌 !

# 질문 및 답변

* max_length를 얼마나 하면 좋을지? 

  - 현재 학생들이 평균적으로 푼 문제는 300개 정도인데 default는 20개여서!

  * (답변) 이전에 NLP competition에서는 Huggingface에서 pretrained된 걸 가져와서 지도학습했음. 이때 max_seq_length는 train data를 기준으로 max에 가까운 값을 지정했던 것 같음 (max와 평균이 차이가 있었음에도 max와 더 가까운 값으로 사용)

* 테스트 데이터 셋을 학습에 이용해도 괜찮은건가?

  * Pseudolabeling 포함 데이터를 학습에 포함시켜도 된다.
  * (답변) 매뉴얼하게 입력하는 것만 아니면 괜찮다.

# 마무리 + 저녁메뉴공유

* 민용: 삼겹살
* 정현: 치킨
* 아라: 초밥
* 유라: 샌드위치 & 비요뜨
* 재훈: (뭘까:question:)
* 진현: (뭘까:question:)

**Comment by MignonDeveloper** [comment]
정성이 듬뿍 느껴진다 ㅜㅡㅜ
